core/src/main/scala/kafka/log/AbstractIndex.scala

```diff
@@ -197,7 +197,9 @@ abstract class AbstractIndex[K, V](@volatile var file: File, val baseOffset: Lon
           /* Windows won't let us modify the file length while the file is mmapped :-( */
           if (OperatingSystem.IS_WINDOWS)
             safeForceUnmap()
-          raf.setLength(roundedNewSize)
+          CoreUtils.retry(3, 200) {
+            raf.setLength(roundedNewSize)
```

Comment:
doesn't this means that the unmap() was not successful? Do we know when this can happen ?

Reply:
Frankly, I do not know if the unmap was not successful. Understanding as of now is that Unmap just got delayed as there was no exception from this call. Retry is to give time if the UnMap is taking longer to finish by OS. Thought process is, we observe this change for few weeks... and if we are still seeing this exception happening, figure out some other plan. Graceful shutdown changes that I have made on the broker side should alleviate some of the problems